### DISCUSSION TITLE
UnknownVariables doesn't return unknown variables defined in files

### DISCUSSION THREAD

**Discussion by ghost** [SCons/scons#4644] [General]
Running `scons` will print `{}` while running `scons my_var="cli_value"` will print `{'my_var': 'cli_value'}`
```python
#custom.py
my_var="value"
```

```python
import SCons
from SCons.Variables import *

env = Environment()

options = Variables(["custom.py"], ARGUMENTS)
options.Add("my_option", "", "")

options.Update(env)

print(options.UnknownVariables())
```

Does it already have separate issue for it?

**Comment by bdbaddog** [comment]
There's no issue for this. Please file.

Looking at the code it's clear it's only checking command line arguments.

**Reply by ghost** [reply]
> Looking at the code it's clear it's only checking command line arguments.

Yeah, I know I've also already looked :slightly_smiling_face: But you asked to be sure if there is no existing issues so I asked here firstly

**Reply by bdbaddog** [reply]
Actually I asked you to follow the process in the bug filing template...

## Please bring your issue to the SCons users mailing list before filing an issue here
## See: https://scons.org/bugs.html
## or join our Discord server : https://discord.gg/bXVpWAy

**Reply by ghost** [reply]
Yeah but if nothing from the above can be really done there is "as close as possible" way should be used which is github discussions (which I saw are duplicated in discord...)

**Reply by bdbaddog** [reply]
I'm not sure "why nothing above can be done?" Can you explain?

**Reply by ghost** [reply]
I mean I **can't** (primary reason) and don't want really write in discord. I don't love mailing lists. But I can ask here...

**Comment by mwichmann** [comment]
This has been addressed now via #4653. I'm going to close this discussion as a result.
